### ISSUE TITLE
Unknown Type Selector doesn't account for ampersands in nested SCSS selector definitions

### ISSUE THREAD

**Issue by derekaug** [racodond/sonar-css-plugin#63]
We have multiple selector definitions nested within our SCSS files that utilize the `&` to refer to the parent class like so:

```
.parent {
    &__child
}
```

where `&__child` would compile to `.parent__child`. The problem is that when we run the scanner, it tells us that `__child` is an unknown type selector. Is there any way to fix this short of turning off this rule?

**Comment by racodond** [comment]
Hi @derekaug,

Could you please provide a valid SCSS example to reproduce the issue?
Which version of the plugin are you running?

Thanks!

Cheers,

David

**Comment by mven** [comment]
I'm also getting this error on valid SCSS declarations. Example would be:

```scss

// example of a BEM-style declaration
.my-block {
  &__my-element {
    &--my-modifier {
    }
  }
}
```

Reference: http://thesassway.com/intermediate/referencing-parent-selectors-using-ampersand

**Comment by racodond** [comment]
Hi @mven,

I cannot reproduce. Could you please provide a screenshot and the version of the CSS plugin?

Thank you!

David

**Comment by mven** [comment]
@racodond I suppose those would be good to have 😄 

`sonar-css-plugin` version installed is `3.1`
SonarQube version is `6.2`

**Comment by racodond** [comment]
OK. You should upgrade to the latest version (4.10):  https://github.com/racodond/sonar-css-plugin/releases/tag/4.10

**Comment by mven** [comment]
K -- I'll try to install that version. Though, would you happen to know why SonarQube doesn't report there's an updated version?

Installed:
![screen shot 2017-10-03 at 11 17 50 am](https://user-images.githubusercontent.com/940113/31136035-900be0b6-a82c-11e7-8ad1-793c2ded64ea.png)

Updates:
![screen shot 2017-10-03 at 11 18 00 am](https://user-images.githubusercontent.com/940113/31136039-92918890-a82c-11e7-983f-1833a5af966d.png)

**Comment by racodond** [comment]
You need to manually install the latest version because versions of the CSS plugin (same for the other plugins I maintain) are no longer available in the Update Center. See https://github.com/racodond/sonar-css-plugin/issues/26.

**Comment by mven** [comment]
Thanks -- I'll try that.

**Comment by mven** [comment]
Updated to the newest version and it looks like I'm not getting the error anymore. Thanks!

**Comment by racodond** [comment]
Thanks for the feedback!
